db/migrate/20181107164348_create_report_type.rb

```diff
@@ -0,0 +1,11 @@
+class CreateReportType < ActiveRecord::Migration[5.0]
+  def change
+    create_table :report_types do |t|
```

Comment:
I'm not sure `report_types` is the best name here. Maybe `reportable_fields`?

This table will store one row per entry in https://github.com/lpichler/manageiq/blob/ce3bb5dc39eb168a70a9f6be73ed496cb5dc6c22/app/models/chargeback_vm.rb#L2 right?

Reply:
first usage, is for ChargeableField. One row of `ChargeableField` could have 2 chargeback report types. [For example, ](https://github.com/ManageIQ/manageiq/pull/18201/files#diff-054944b30b52bd84979beef99d202b0aR9) `v_derived_cpu_total_cores_used`  belongs to `ChargebackContainerImage, ChargebackContainerProject` chargeback report types.

So firstly I was modeling: 

 `ChargeableField` **has_many** `ChargebackReportType`  and then
I realized that probably it could be useful to don't limit ourselves and go with general way:

`ChargeableField` **has_many** `ReportType`

due this facts, I think that name `reportable_fields` doesn't explain the idea of intended report types. But as my native language is not english -  I could be wrong :) 

what do you think now?

thanks!

Reply:
I'm still not sure what the resource is pointing to in that case? If its something inheriting from `Chargeback` then there won't be a database row, but if it's `ChargeableField` then why make it polymorphic?